### ISSUE TITLE
RODECASTER PRO/RODECASTER PRO II Module

### ISSUE THREAD

**Issue by BaileyStoneMedia** [bitfocus/companion-module-requests#885]
Hi there,

I am working with a RODE RODECASTER in an upcoming production and would like to be able to do the following:
- Stop/Start Recording on Device
- Mute/Unmute Specific Inputs on Device (Eg. Mute XLR 1)
- Control Faders
- Trigger built-in sounds similar to sound pads on RODECASTER device
- Control headphone and output volume

Unfortunately the original RODECASTER Pro does not have an ethernet port, so it would have to be connected via USB, so not sure if that would work. 

Thanks for making such an awesome piece of software that assists with so many productions

Kind Regards
Bailey Stone
Owner - Bailey Stone Media

**Comment by josephdadams** [comment]
Since it has no ethernet port and no documented protocol, probably the best solution in this case would be some kind of middle-man app that has the necessary logic in it to work directly with the USB connection for that device, and that app then exposes a bi-directional network protocol for Companion. There are several modules like this where someone has created a bespoke middle-man program to handle that translation.

**Comment by josephdadams** [comment]
It would have to be custom made for you.

**Comment by BaileyStoneMedia** [comment]
Ahh ok, thanks for letting me know

**Comment by josephdadams** [comment]
Do you know if the app that installs with the Rodecaster allows for the type of control you are looking for? Not to be confused with Companion, but they do call their app the "Companion App".

**Comment by BaileyStoneMedia** [comment]
No, unfortunately the software that comes with it doesn't have that control

**Comment by progressmedia** [comment]
This is flagged "blocked externally". Is this really the case? I don't see anyone saying it's blocked.

**Comment by josephdadams** [comment]
> Since it has no ethernet port and no documented protocol, probably the best solution in this case would be some kind of middle-man app that has the necessary logic in it to work directly with the USB connection for that device, and that app then exposes a bi-directional network protocol for Companion. There are several modules like this where someone has created a bespoke middle-man program to handle that translation.

@progressmedia this is why.

**Comment by nastasec89** [comment]
Maybe we can look into making something for Pro II as it has an ethernet port.. ??? Maybe...

**Comment by progressmedia** [comment]
Yes I agree. The procaster II has an ethernet port. Any thoughts @josephdadams?

**Comment by josephdadams** [comment]
Share the API/protocol for it

**Comment by T-Staal** [comment]
I also  working with a RODE RODECASTER 2  and ther is a ethernet port on it  and also connet on usb-c.

i like to use RØDE. whit companion.
-Thomas

**Comment by progressmedia** [comment]
I can't find the API/protocols online or in the manual, but I have to believe that this is possible. If you look at Rode's marketing materials, you'll see that they talk about using MIDI triggers through the Rodecaster Pro II's smart pads. I'll reach out to Rode and see if they have released anything. 
https://rode.com/en-us/user-guides/rodecaster-pro-ii/using-the-smart-pads

This is for the Rodecaster Pro I (NOT the II), but it looked like they are on the right path in terms of using a packet sniffing USB type of setup through pacmd?
https://forum.level1techs.com/t/rodecaster-pro-bad-audio-out-solved-ish/170601


Alternatively here is some documentation about MIDI control with the rodecaster pro II
https://cdn1.rode.com/rodecasterpro/v2.1.2_beta/MIDI+Walkthroughs/Using+MIDI+to+control+DAW-Reaper+Mac.pdf
https://rode.com/en-us/user-guides/rodecaster-pro/midi-control

**Comment by jasonesg** [comment]
Hey @josephdadams, I have a RodeCaster Pro 2 and I'm interested in using it with the companion app. I'm particularly interested in the following:

1. Stop/Start Recording on the RodeCaster Pro 2
2. Mute/Unmute Specific Inputs on the RodeCaster Pro 2
4. MIDI Soundboard
5. Exporting shows

I'd appreciate it if you could let me know if these features are currently available with the companion app with the RodeCaster Pro 2. If not, would it be possible for you to add these features in a future update?

Thanks a lot, hope to hear from you soon!

**Comment by josephdadams** [comment]
There is not currently a rodecaster module in companion.

**Comment by jasonesg** [comment]
> There is not currently a rodecaster module in companion.

Is it possible to request a Rodecaster module for companion considering there are a handful of people looking for one across Reddit/Github/Quora and etc?

**Comment by josephdadams** [comment]
That’s what this thread is.

It’s going to take a developer serious time since there is no API available. The whole thing will need to be reverse engineered.

on the other hand, if you can do it with midi, you could do it now with companion + midi-relay.

**Comment by 4ThePerks** [comment]
I am interested in this too as I have just purchased a Rodecaster Pro II that will replace the first model.  The companion app is very helpful.  Has there been any progress towards having control of the Rodecaster Pro II through companion?

**Comment by progressmedia** [comment]
> I am interested in this too as I have just purchased a Rodecaster Pro II that will replace the first model.  The companion app is very helpful.  Has there been any progress towards having control of the Rodecaster Pro II through companion?
> 
> 

No, Rode has never even talked about it.

**Comment by jjmmfi** [comment]
I found this video that explains how you might be able to control some of the Rodecaster settings with midi controls: https://www.youtube.com/watch?v=Eey5NawNKWw

**Comment by josephdadams** [comment]
Closing as **not planned**.

This request is labelled as having no external control protocol (or as unlikely to be feasible) — the device or software doesn't expose an interface Companion could talk to, so a module can't be built for it as things stand.

If the manufacturer adds a control API later, please comment and we'll happily reopen.
